nft-client/src/main.rs

```diff
@@ -0,0 +1,635 @@
+                for event in events {
+                    let mut cursor = concordium_contracts_common::Cursor::new(event.as_ref());
+                    let event = cts1::Event::deserial(&mut cursor)
+                        .map_err(|_| anyhow!("Failed parsing event"))?;
+                    println!("{}", event);
+                }
+            }
+        }
+        Command::Mint {
+            sender,
+            token_ids,
+            contract,
+        } => {
+            // The GRPC client for Concordium node.
+            let mut grpc_client = Client::connect(app.endpoint, "rpcadmin".to_string())
+                .await
+                .context("Failed to connect to Node GRPC")?;
+
+            let account_data: AccountData = serde_json::from_str(
+                &std::fs::read_to_string(sender).context("Could not read the keys file.")?,
+            )
+            .context("Could not parse the accounts file.")?;
+
+            let owner = convert_account_address(&account_data.address);
+            let mint_parameter = cts1::MintParams {
+                owner:     concordium_contracts_common::Address::Account(owner),
+                token_ids: token_ids.clone(),
+            };
+            let bytes = concordium_contracts_common::to_bytes(&mint_parameter);
+
+            let update_payload = transactions::UpdateContractPayload {
+                amount:       common::types::Amount::from(0),
+                address:      contract.0,
+                receive_name: smart_contracts::ReceiveName::try_from(format!(
+                    "{}.mint",
+                    NFT_CONTRACT_NAME
+                ))
+                .map_err(|e| anyhow!("Failed to parse receive name {}", e))?,
+                message:      smart_contracts::Parameter::from(bytes),
+            };
+
+            let transaction_payload = transactions::Payload::Update {
+                payload: update_payload,
+            };
+            print!("Minting tokens with id: ");
+            for token_id in token_ids {
+                print!("{} ", token_id);
+            }
+            println!();
+            let hash =
+                send_transaction(&mut grpc_client, &account_data, transaction_payload).await?;
+            println!("Transaction send {}", hash);
+        }
+        Command::Transfer {
+            sender,
+            token_ids,
+            from,
+            to,
+            to_func,
+            to_func_data,
+            contract,
+        } => {
+            // The GRPC client for Concordium node.
+            let mut grpc_client = Client::connect(app.endpoint, "rpcadmin".to_string())
+                .await
+                .context("Failed to connect to Node GRPC")?;
+
+            let account_data: AccountData = serde_json::from_str(
+                &std::fs::read_to_string(sender).context("Could not read the keys file.")?,
+            )
+            .context("Could not parse the accounts file.")?;
```

Comment:
Recommend putting this before establishing the connection to not waste resources.